cypress/pages/BounceRulesContainer/index.js

```diff
@@ -157,9 +157,22 @@ class BounceRulesPage extends Page {
     return this.confirmationSubmit.click();
   }
 
-  deleteBounceRuleUI(id) {
-    this.createdRuleButton(id).click();
-    return this.deleteConfirmationConfirm.click();
+  deleteBounceRuleUI(testRule) {
+    cy.task("getRules").then(res => {
+      for (let i = 0; i < res.length; i++) {
```

Comment:
use built in map func is preferable since for loop with index i iterated over doesn’t show as much intent right away

Reply:
another benefit is that they’re functional, meaning there are no side effects or variables possibly being changed. when you call it on an array it returns a new array. same input/output every time. it doesn’t modify the existing array

Reply:
with a for loop you can do anything like edit the existing array and all that. overall it’s cleaner and more understandable code at first glance from the maintainer perspective